File: node/1425.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

for(line of lines) {
    let [stones, gift] = line.split(" ")
    
    if (stones == "0" && gift == "0") {
        return
    } 

    if (gift > 34) {
        console.log("Let me try")
    } else {
        stones = parseInt(stones)
        gift = parseInt(gift)
    
        let queue = []
        let win = false
    
        queue.push([2, 1])
    
        while (queue.length) {
            let [currentJump, currentStone] = queue.shift()
    
            if (currentStone == gift) {
                win = true
            }
    
            let nextJumpSize = currentJump * 2 - 1
            currentJump += 1
    
            let nextJumpAhead = currentStone + nextJumpSize
            let nextJumpBack = currentStone - nextJumpSize
    
            if (nextJumpAhead <= stones) {
                queue.push([currentJump, nextJumpAhead])
            }
    
            if (nextJumpBack > 0) {
                queue.push([currentJump, nextJumpBack])
            }
        }
    
        console.log(win ? "Let me try!" : "Don't make fun of me!")
    }
}

File: python/1425.py

while 1:
    stones, gift = input().split(" ")
    stones, gift = int(stones), int(gift)

    if stones == 0 and gift == 0:
        break

    queue = []

    queue.append([1, 1])

    win = 0

    while queue:
        current_jump, curr_stone = queue.pop(0)

        if curr_stone < 1 or curr_stone > stones:
            continue

        if curr_stone == gift:
            win = 1
            break

        current_jump += 1
        next_jump_size = current_jump * 2 - 1
        
        next_stone_jump_ahead = curr_stone + next_jump_size
        next_stone_jump_back = curr_stone - next_jump_size

        queue.append([current_jump, next_stone_jump_ahead])
        queue.append([current_jump, next_stone_jump_back])

        
    print("Let me try!" if win else "Don't make fun of me!")

